src/main/java/uk/gov/hmcts/darts/arm/mapper/impl/MediaArchiveRecordMapperImpl.java

```diff
@@ -1,89 +1,256 @@
+            metadata.setBf014(mapToInt(mediaRecordProperties.getProperty(BF_014_KEY), media));
+        }
+        if (mediaRecordProperties.containsKey(BF_015_KEY)) {
+            metadata.setBf015(mapToInt(mediaRecordProperties.getProperty(BF_015_KEY), media));
+        }
+        if (mediaRecordProperties.containsKey(BF_016_KEY)) {
+            metadata.setBf016(mapToString(mediaRecordProperties.getProperty(BF_016_KEY), media));
+        }
+        if (mediaRecordProperties.containsKey(BF_017_KEY)) {
+            metadata.setBf017(mapToString(mediaRecordProperties.getProperty(BF_017_KEY), media));
+        }
+        if (mediaRecordProperties.containsKey(BF_018_KEY)) {
+            metadata.setBf018(mapToString(mediaRecordProperties.getProperty(BF_018_KEY), media));
+        }
+        if (mediaRecordProperties.containsKey(BF_019_KEY)) {
+            metadata.setBf019(mapToString(mediaRecordProperties.getProperty(BF_019_KEY), media));
+        }
+        if (mediaRecordProperties.containsKey(BF_020_KEY)) {
+            metadata.setBf020(mapToString(mediaRecordProperties.getProperty(BF_020_KEY), media));
         }
         return metadata;
     }
 
+    private String mapToString(String key, MediaEntity media) {
+        return switch (key) {
+            case OBJECT_TYPE_KEY -> ArchiveRecordType.MEDIA_ARCHIVE_TYPE.getArchiveTypeDescription();
+            case CASE_NUMBERS_KEY -> getCaseNumbers(media);
+            case FILE_TYPE_KEY -> media.getMediaFormat();
+            case HEARING_DATE_KEY -> getHearingDate(media);
+            case CHECKSUM_KEY -> media.getChecksum();
+            case CREATED_DATE_TIME_KEY -> formatDateTime(media.getCreatedDateTime());
+            case START_DATE_TIME_KEY -> formatDateTime(media.getStart());
+            case END_DATE_TIME_KEY -> formatDateTime(media.getEnd());
+            case COURTHOUSE_KEY -> getCourthouse(media);
+            case COURTROOM_KEY -> getCourtroom(media);
+            default -> null;
+        };
+    }
+
+    private String formatDateTime(OffsetDateTime offsetDateTime) {
+        String dateTime = null;
+        if (nonNull(offsetDateTime)) {
+            dateTime = offsetDateTime.format(dateTimeFormatter);
+        }
+        return dateTime;
+    }
+
+    private String getHearingDate(MediaEntity media) {
+        String hearingDate = null;
+        if (CollectionUtils.isNotEmpty(media.getHearingList())) {
+            hearingDate = media.getHearingList().get(0).getHearingDate().format(dateFormatter);
+        }
+        return hearingDate;
+    }
+
+    private String getCaseNumbers(MediaEntity media) {
+        String cases = null;
+        if (nonNull(media.getCaseNumberList())) {
+            cases = caseListToString(media.getCaseNumberList());
+        }
+        return cases;
+    }
+
+    private static String getCourthouse(MediaEntity media) {
+        String courthouse = null;
+        if (nonNull(media.getCourtroom()) && nonNull(media.getCourtroom().getCourthouse())) {
```

Comment:
it shouldn't be possible to have a courtroom without a courthouse as there is a database constraint, so i guess the 2nd part of the IF would only be applicable for unit tests where test data hasn;t been setup correctly?